tests/test_converters.py

```diff
@@ -184,3 +185,78 @@ def test_duration_converter_for_invalid(duration: str):
     converter = Duration()
     with pytest.raises(BadArgument, match=f'`{duration}` is not a valid duration string.'):
         asyncio.run(converter.convert(None, duration))
+
+
+@pytest.mark.parametrize(
+    ("datetime_string", "expected_dt"),
+    (
+
+        # `YYYY-mm-ddTHH:MM:SSZ` | `YYYY-mm-dd HH:MM:SSZ`
+        ('2019-09-02T02:03:05Z', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02 02:03:05Z', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+
+        # `YYYY-mm-ddTHH:MM:SS±HH:MM` | `YYYY-mm-dd HH:MM:SS±HH:MM`
+        ('2019-09-02T03:18:05+01:15', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02 03:18:05+01:15', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02T00:48:05-01:15', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02 00:48:05-01:15', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+
+        # `YYYY-mm-ddTHH:MM:SS±HHMM` | `YYYY-mm-dd HH:MM:SS±HHMM`
+        ('2019-09-02T03:18:05+0115', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02 03:18:05+0115', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02T00:48:05-0115', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02 00:48:05-0115', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+
+        # `YYYY-mm-ddTHH:MM:SS±HH` | `YYYY-mm-dd HH:MM:SS±HH`
+        ('2019-09-02 03:03:05+01', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02T01:03:05-01', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+
+        # `YYYY-mm-ddTHH:MM:SS` | `YYYY-mm-dd HH:MM:SS`
+        ('2019-09-02T02:03:05', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+        ('2019-09-02 02:03:05', datetime.datetime(2019, 9, 2, 2, 3, 5)),
+
+        # `YYYY-mm-ddTHH:MM` | `YYYY-mm-dd HH:MM`
+        ('2019-11-12T09:15', datetime.datetime(2019, 11, 12, 9, 15)),
+        ('2019-11-12 09:15', datetime.datetime(2019, 11, 12, 9, 15)),
+
+        # `YYYY-mm-dd`
+        ('2019-04-01', datetime.datetime(2019, 4, 1)),
+
+        # `YYYY-mm`
+        ('2019-02-01', datetime.datetime(2019, 2, 1)),
+
+        # `YYYY`
+        ('2025', datetime.datetime(2025, 1, 1)),
+    ),
+)
+def test_isodatetime_converter_for_valid(datetime_string: str, expected_dt: datetime.datetime):
+    converter = ISODateTime()
+    assert asyncio.run(converter.convert(None, datetime_string)) == expected_dt
```

Comment:
Does the equality comparison in this assertion already cover checking that the returned `datetime` is timezone-naïve?

Reply:
Good point, I've added an explicit assertion to check that.